### PR TITLE
Dpg 1671 edit master data / disable+enable master data

### PR DESCRIPTION
**Pull request by nipunarora-eGov**

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js [hunk 1]
```diff
@@ -28,12 +28,12 @@ const MDMSAdd = ({ defaultFormData, updatesToUISchema, screenType = "add", onVie
...
   const [loadDependent, setLoadDependent] = useState([]);
   const [showErrorToast, setShowErrorToast] = useState(false);
-
+  
   const [showToast, setShowToast] = useState(false);
   const { moduleName, masterName } = Digit.Hooks.useQueryParams();
-
+  
   useEffect(() => {
-    setSession({ ...session, ...defaultFormData });
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:36]: why session was removed ? @nipunarora-eGov

Comment (nipunarora-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:36]: Initially updating with the default values(data from server). This session  state had data of some previous master and was populating incorrectly. @jagankumar-egov

Comment (jagankumar-egov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:36]: check again reverting that now?

Comment (nipunarora-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:36]: Updated @jagankumar-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js [hunk 2]
```diff
@@ -43,7 +44,7 @@ const MDMSAdd = ({ defaultFormData, updatesToUISchema, screenType = "add", onVie
     params: {},
     body: {
       SchemaDefCriteria: {
-        tenantId: tenantId || stateId,
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:46]: tenantid has more priority , so when a useer at city a , he adds data to city a
and in search screen he sees data in city a

Comment (nipunarora-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSAddV2.js:46]: updated

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/configs/UICustomizations.js [hunk 3]
```diff
@@ -428,10 +428,9 @@ export const UICustomizations = {
     },
     preProcess: (data,additionalDetails) => {
       
-      const tenant = Digit.ULBService.getStateId();
-      data.body.MdmsCriteria.tenantId = tenant
-      
-      data.body.MdmsCriteria.tenantId = Digit.ULBService.getStateId();
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/configs/UICustomizations.js:434]: this is schema search right ?
Here it is working with this change ?

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSEdit.js [hunk 4]
```diff
@@ -25,12 +29,72 @@ const MDMSEdit = ({...props}) => {
...
+    );
+
+  }
+
   if(isLoading) return <Loader />
 
   return (
-    <MDMSAdd defaultFormData = {data?.data} screenType={"edit"} />
+    <React.Fragment>
+      <MDMSAdd defaultFormData = {data?.data} screenType={"edit"} onSubmitEditAction={handleUpdate}  />
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSEdit.js:95]: unique fields can't be edited

Comment (nipunarora-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/workbench/src/pages/employee/MDMSEdit.js]: updated

**Review by jagankumar-egov — APPROVED** [review]

**Pull request merged**